oauth-server/src/main/java/io/strimzi/kafka/oauth/server/JaasServerOauthValidatorCallbackHandler.java

```diff
@@ -113,6 +113,19 @@ public void configure(Map<String, ?> configs, String saslMechanism, List<AppConf
         }
     }
 
+    @SuppressWarnings("deprecation")
+    private static boolean isCheckAccessTokenType(Config config) {
+        String legacy = config.getValue(ServerConfig.OAUTH_VALIDATION_SKIP_TYPE_CHECK);
+        if (legacy != null) {
+            log.warn("OAUTH_VALIDATION_SKIP_TYPE_CHECK is deprecated. Use OAUTH_CHECK_ACCESS_TOKEN_TYPE (with reverse meaning) instead.");
+            if (config.getValue(ServerConfig.OAUTH_CHECK_ACCESS_TOKEN_TYPE) != null) {
+                throw new RuntimeException("OAuth validator configuration error: can't use both OAUTH_CHECK_ACCESS_TOKEN_TYPE and OAUTH_VALIDATION_SKIP_TYPE_CHECK");
+            }
+        }
+        return legacy != null ? !Config.isTrue(legacy) :
```

Comment:
Same as above ... Is it really correct that the legacy option is prefered over the new one? Shouldn't it be the other way around? Or does it have some other reason?

Reply:
Same answer - this code won't be reached if both are present.